Automattic/wp-calypso
# Add logic for site_edited task

### Proposed Changes

This PR adds link and task completion logic for the 'Edit site design' task on Launchpad for the Free Flow. The following specific changes are being made: 
  * The 'Edit your site' task now links to the Site Editor
  * The 'Edit your site' task will show correct completion status (ie, will show as completed once completed)

**Backend diff: D94910-code** This PR has an associated backend diff that updates the site_edited option when users in the free flow visit the Site Editor. 
  
### Testing Instructions

Test the Free Flow and confirm the Edit your site task works as expected:
   - Go to this URL to start the free flow: http://calypso.localhost:3000/setup/free/intro
   - Go through the flow until Launchpad
   - Sandbox the site you just created
   - TEST: The 'Edit your site' item should show as incomplete, should be clickable, and should link to the Site Editor. 
   - TEST: The 'Launch your site' button should be disabled.
   - Follow the link to the Site Editor
   - Return to Launchpad. Normally you could do this by just clicking the Go to Dashboard links, which goes to My Home, which redirects to Launchpad. However, the back to dashboard button currently will redirect you from http://calypso.localhost:3000/ to https://wordpress.com. I think this is due to a recent change. Once the urls is wordpress.com, our feature flag will not allow the launchpad screen to load and you'll be redirected to the old free flow. To get around all this, you should either use the back button to go Launchpad or you'll need to manually re-enter the correct launchpad for your local dev url, which is: `http://calypso.localhost:3000/setup/link-in-bio/launchpad?siteSlug=YOURSITESLUG`
   - TEST: When you return the Launchpad, the 'Edit your site' item should show as complete (but still be clickable)
   - TEST: The 'Launch your site' button should now be active

### Other

Addresses: https://github.com/Automattic/wp-calypso/issues/70558
